.github/workflows/turborepo-test-go-fallback.yml

```diff
@@ -0,0 +1,145 @@
+name: Turborepo Test (Go Fallback)
+on:
+  push:
+    branches: [main]
+    paths:
+      - .cargo/**
+      - Cargo.*
+      - Cargo.**
+      - cli/**
+      - crates/turborepo*/**
+      - package.json
+      - pnpm-lock.yaml
+      - rust-toolchain
+      - shim/**
+      - turborepo-tests/helpers/**
+      - turborepo-tests/integration/**
+      - version.txt
+      - xtask/**
+  pull_request:
+    paths:
+      - .cargo/**
+      - Cargo.*
+      - Cargo.**
+      - cli/**
+      - crates/turborepo*/**
+      - package.json
+      - pnpm-lock.yaml
+      - rust-toolchain
+      - shim/**
+      - turborepo-tests/helpers/**
+      - turborepo-tests/integration/**
+      - version.txt
+      - xtask/**
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
+
+permissions:
+  actions: write
+  contents: read
+  pull-requests: read
+
+jobs:
+  lint:
+    name: Lint
+    runs-on:
+      - "self-hosted"
+      - "linux"
+      - "x64"
+      - "metal"
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Setup Go
+        uses: ./.github/actions/setup-go
+        with:
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+
+      - name: Setup Rust
+        uses: ./.github/actions/setup-rust
+
+      - name: Setup Protos
+        run: cd cli && make compile-protos
+
+      - name: Build turborepo-ffi
+        run: cd cli && make turborepo-ffi-install
+
+      - name: golangci
+        uses: golangci/golangci-lint-action@v3
+        with:
+          # More config options documented here: https://github.com/golangci/golangci-lint-action
+          version: latest
+          working-directory: cli
+          args: --new-from-rev=${{ github.event.pull_request.base.sha || 'HEAD~1' }}
+
+  unit:
+    name: Unit Tests
+    runs-on: ${{ matrix.os.runner }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - name: ubuntu
+            runner:
+              - "self-hosted"
+              - "linux"
+              - "x64"
+              - "metal"
+          - runner: macos-latest
+          - runner: windows-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/setup-turborepo-environment
+        with:
+          windows: ${{ matrix.os.runner == 'windows-latest' }}
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+
+      - run: turbo run test --filter=cli --color
+        env:
+          EXPERIMENTAL_RUST_CODEPATH: true
```

Comment:
We can drop this now that we're on a version of `turbo` that uses Rust by default

```suggestion
```